engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/event/StartEventProcessor.java

```diff
@@ -43,13 +44,19 @@ public Class<ExecutableStartEvent> getType() {
   }
 
   @Override
-  public void onActivate(final ExecutableStartEvent element, final BpmnElementContext context) {
+  public void onActivate(
+      final ExecutableStartEvent element,
+      final BpmnElementContext context,
+      final SideEffects sideEffects) {
     final var activated = stateTransitionBehavior.transitionToActivated(context);
     stateTransitionBehavior.completeElement(activated);
   }
 
   @Override
-  public void onComplete(final ExecutableStartEvent element, final BpmnElementContext context) {
+  public void onComplete(
+      final ExecutableStartEvent element,
+      final BpmnElementContext context,
+      final SideEffects sideEffects) {
```

Comment:
Could also add a separate method which is called by the default impl, without the side-effects for implementations which don't need this - this would allow to change less code. and makes it cleary that only if yo need it implement that method x

Reply:
Alternative we could extend the Writers class, to be able to write side effects or something. Maybe Writers need to renamed then as well. Right now Writers class is an abstraction around the result builder used in the engine